File: project_euler/Problem099/Problem099.go
package project_euler

import (
    "bufio"
    "fmt"
    "io"
    "math"
    "os"
    "strconv"
    "strings"
)

func Problem099() {
    fileName := "base_exp.txt"
    file, _ := os.Open(fileName)
    handle := bufio.NewReader(file)
    index := 1
    pivot := float64(0)
    result := 0
    for true {
        line, _, err := handle.ReadLine()
        if io.EOF == err {
            break
        }

        elements := strings.Split(strings.Replace(string(line), "\n", "", -1), " ")
        auxiliary, _ := strconv.Atoi(elements[0])
        value := math.Log(float64(auxiliary))
        auxiliary, _ = strconv.Atoi(elements[1])
        value *= float64(auxiliary)
        if pivot < value {
            pivot = value
            result = index
        }
        index++
    }
    file.Close()
    fmt.Println(result)
}

File: project_euler/Problem099/Problem099.js
function Problem099() {
    let fileName = 'base_exp.txt'
    const file = require('n-readlines')
    let handle = new file(fileName)
    let line
    let index = 1
    let pivot = 0
    let result = 0
    while (line = handle.next()) {
        let elements = line.toString().split(' ')
        let value = Math.log(Number(elements[0])) * Number(elements[1])
        if (pivot < value) {
            pivot = value
            result = index
        }
        ++index
    }
    console.log(result)
}

module.exports = {
    Problem099 : Problem099
};
